File: Pattern37.c
/**
 * C program to print left arrow star pattern
 */

#include <stdio.h>

int main()
{
    int i, j, n;

    // Input number of rows from user
    printf("Enter value of n : ");
    scanf("%d", &n);

    // Print upper part of the arrow
    for(i=1; i<n; i++)
    {
        // Print trailing (n-rownumber) spaces
        for(j=1; j<=(n-i); j++)
        {
            printf(" ");
        }

        // Print inverted right triangle
        for(j=i; j<=n; j++)
        {
            printf("*");
        }

        printf("\n");
    }

    // Print bottom part of the arrow
    for(i=1; i<=n; i++)
    {
        // Print trailing (rownumber-1) spaces
        for(j=1; j<i; j++)
        {
            printf(" ");
        }

        // Print the right triangle
        for(j=1; j<=i; j++)
        {
            printf("*");
        }

        printf("\n");
    }

    return 0;
}

File: Pattern37.py


no = int(input("Enter Star Dimension: "))
print("-------------------------------------")


rows=1


totColumns=(no*2)-1
auxColumns=totColumns

while (rows <= totColumns):
    if(rows <=no ): 
        j=totColumns-no-rows+1
        for i in range(0,j):
          print("",end=' ')
        for i in range(j,auxColumns):
          print("*",end='')
        print()
        auxColumns-=2
        if(rows==no):
           auxColumns+=2
    else:
       auxColumns+=2
       j=rows-no
       for i in range(0,j):
          print("",end=' ')
       for i in range(j,auxColumns):
          print("*",end='')
       print()
    rows+=1

print("-------------------------------------")